msteams-platform/tabs/how-to/dashboard-teams-toolkit.md

```diff
@@ -0,0 +1,991 @@
+    return undefined;
+  }
+
+  /**
+   * Override this method to customize the widget body.
+   * @returns JSX component for the widget body
+   */
+  protected bodyContent(): JSX.Element | undefined {
+    return undefined;
+  }
+
+  /**
+   * Override this method to customize the widget footer.
+   * @returns react node for the widget footer
+   */
+  protected footerContent(): JSX.Element | undefined {
+    return undefined;
+  }
+}
+```
+
+Following are the recommended methods to override:
+
+| Methods | Function | Recommend to override |
+|---|---|---|
+| **constructor()** | Invokes the initial `this.state` and call the constructor of the super class `React` Component. | NO |
+| **componentDidMount()** | Invokes after a component is mounted and assigns a value to the `data` property of the state by calling the `getData()` method. | NO |
+| **render()** | Invokes each time a update happens and the dashboard default layout is defined in this method. | NO |
+| **getData()** | Invokes the data needed by the widget, and the value returned by this method will be set to `this.state.data`. |
+| **headerContent()** | Invokes what the widget header will look like. You can choose to override this method to customize a widget or not, if not, the widget won't have a header. | YES |
+| **bodyContent()** | Invokes what the widget body will look like. You can choose to override this method to customize a widget or not, if not, the widget won't have a body. | YES |
+| **footerContent()** | Invokes what the widget footer will look like. You can choose to override this method to customize a widget or not, if not, the widget won't have a footer. | YES |
+
+### Microsoft Graph Toolkit as widget content
+
+Microsoft Graph Toolkit is a set of renewable, framework-agnostic web component which helps accessing and working with Microsoft Graph. You can use the Microsoft Graph Toolkit with any web framework or without a framework.
+
+You can follow the steps to use Microsoft Graph Toolkit as your widget content:
+
+1. Add SSO feature to your Teams app
+
+   Microsoft Teams provides single sign-on (SSO) function for an app to obtain signed in Teams user token to access Microsoft Graph.
+
+   For more information, refer [SSO feature to your Teams app](../../toolkit/add-single-sign-on.md).
+
+1. Install required `npm` packages
+
+   Run the following command in your project `tabs` folder to install the required `npm` packages:
+
+      ```typescript
+      npm install @microsoft/mgt-react @microsoft/mgt-teamsfx-provider
+      ```
+
+1. Add a new Graph Toolkit widget
+
+      Create a new widget file in your project tabs/src/views/widgets folder. For example, **GraphyWidget.tsx**. In this widget, we'll guide users to consent our app to access Microsoft Graph and then show the user's `Todo` list by using Microsoft Graph Toolkit.
+
+      The following code is an example of using `Todo` component from Microsoft Graph Toolkit in widget:
+
+      ```typescript
```

Comment:
It may be better if the language of this code snippet is adjusted to `tsx`.

Reply:
Updated to `tsx`